OpenTrons/opentrons-api
# feat(app, app-shell-odd): add function to read csv file from usb

# Overview
add a function that loads csv files from an usb-drive, update protocol card & pinned protocol, and add ChooseCsvFile component (display csv files on USB).

close [AUTH-460](https://opentrons.atlassian.net/browse/AUTH-460), [AUTH-468](https://opentrons.atlassian.net/browse/AUTH-468), and partially [AUTH-462](https://opentrons.atlassian.net/browse/AUTH-462) and [AUTH-470](https://opentrons.atlassian.net/browse/AUTH-470)

This PR doesn't update protocol details screen and Parametes screen. They will be implemented in a following pr since this PR's main part is displaying csv files on USB.

The following yellow background shows up when turning on the feature flag and a protocol name includes RTP.
The rendering condition part will be update in a following pr.

CSV files on Robot are mock data.

the following video includes un-plug and re-plug

https://github.com/Opentrons/opentrons/assets/474225/0cc84d4b-4fcb-4cb6-9191-51023cb8d082





# Test Plan
1. send a protocol that requires a csv file to ODD
    check that the protocol card has yellow background and chip in Protocol dashboard
2. tap the protocol card
    there is requires csv text in Protocol details
3. tap set up a run
    there is choose csv file button (yellow background button for this PR) in the top in parameters screen
4. tap choose csv file button
    there are two mock csv file names on Robot
5. create a few csv files (not needed any data) and move them to a USB flash drive then plug into a Flex
    the csv files on the USB flash drive shows up
     


# Changelog
- add register to support csv file reading function
- update protocol card component and its tests to support csv file
- update pinned protocol card component and its tests to support csv file
- add ChooseCSV component and its tests to support csv file
- update protocolsetup parameters component to support csv file


# Review requests



# Risk assessment
low



[AUTH-460]: https://opentrons.atlassian.net/browse/AUTH-460?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[AUTH-468]: https://opentrons.atlassian.net/browse/AUTH-468?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[AUTH-462]: https://opentrons.atlassian.net/browse/AUTH-462?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[AUTH-470]: https://opentrons.atlassian.net/browse/AUTH-470?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
